src/robot/Robot.cpp

```diff
@@ -1,21 +1,454 @@
+  {
+    auto dofs = subrobot.second->mDofs;
+    for (std::string dofName : util::dofNamesFromSkeleton(metaSkeleton))
+    {
+      if (dofs.find(dofName) != dofs.end())
+      {
+        dtwarn << "Subrobot '" << name << "'' overlaps existing subrobot "
+               << subrobot.first << "." << std::endl;
+        return nullptr;
+      }
+    }
+  }
+
+  // Create the subrobot.
+  auto subRobot = std::make_shared<Robot>(metaSkeleton, name);
+  const auto& root = mRootRobot ? mRootRobot : RobotPtr(this);
+  subRobot->setRootRobot(root);
+  subRobot->setWorld(nullptr); // inherit root robot world
+  mSubRobots[name] = subRobot;
+  return subRobot;
+}
+
+//=============================================================================
+Eigen::VectorXd Robot::getNamedConfiguration(const std::string& name) const
+{
+  auto configuration = mNamedConfigurations.find(name);
+  if (configuration == mNamedConfigurations.end())
+    return Eigen::VectorXd();
+
+  return configuration->second;
+}
+
+//=============================================================================
+void Robot::setNamedConfigurations(
+    std::unordered_map<std::string, const Eigen::VectorXd> namedConfigurations)
+{
+  mNamedConfigurations = std::move(namedConfigurations);
+}
+
+//=============================================================================
+trajectory::TrajectoryPtr Robot::planToConfiguration(
+    const Eigen::VectorXd& goalConf,
+    const constraint::TestablePtr& testableConstraint,
+    const std::shared_ptr<planner::ConfigurationToConfigurationPlanner>&
+        planner,
+    const std::shared_ptr<aikido::planner::TrajectoryPostProcessor>
+        trajPostProcessor) const
+{
+  statespace::dart::MetaSkeletonStateSpace::State* goalState = nullptr;
+  try
+  {
+    mStateSpace->convertPositionsToState(goalConf, goalState);
+  }
+  catch (const std::exception& e)
+  {
+    dtwarn << "Cannot convert configuration to robot state: " << e.what()
+           << std::endl;
+    return nullptr;
+  }
+
+  // Create the problem.
+  auto problem = planner::dart::ConfigurationToConfiguration(
+      mStateSpace,
+      mStateSpace->getScopedStateFromMetaSkeleton(mMetaSkeleton.get()),
+      goalState,
+      testableConstraint);
+
+  // Default to base Snap Planner
+  auto basePlanner = planner;
+  if (!basePlanner)
+  {
+    using planner::SnapConfigurationToConfigurationPlanner;
+    using statespace::GeodesicInterpolator;
+    basePlanner = std::make_shared<SnapConfigurationToConfigurationPlanner>(
+        mStateSpace, std::make_shared<GeodesicInterpolator>(mStateSpace));
+  }
+
+  // Convert to DART Planner
+  using planner::dart::
+      ConfigurationToConfiguration_to_ConfigurationToConfiguration;
+  auto dartPlanner = std::make_shared<
+      ConfigurationToConfiguration_to_ConfigurationToConfiguration>(
+      basePlanner, mMetaSkeleton);
+
+  // Solve the problem with the DART planner.
+  auto rawPlan = dartPlanner->plan(problem, /* result */ nullptr);
```

Comment:
I've always been confused as to why result is commented out (even in earlier versions of AIKIDO). Will we ever have to revert to it? If not, perhaps we should delete it (in this line and others?)